src/Backups/BackupsWorker.cpp

```diff
@@ -183,6 +173,17 @@ OperationID BackupsWorker::startMakingBackup(const ASTPtr & query, const Context
     String backup_name_for_logging = backup_info.toStringForLogging();
     try
     {
+        /// Check if there are no concurrent backups
+        if (num_active_backups && !allow_concurrent_backups)
```

Comment:
Maybe we can organize those checks as a separate function?

Reply:
addressed it